unit-tests/py/rspy/unittest.py

```diff
@@ -0,0 +1,367 @@
+        #test:device L500* D400*
+        #test:retries 3
+        #test:priority 0
+    And, for C++ the prefix could be:
+        //#test:...
+    """
+
+    def __init__( self, source, line_prefix ):
+        """
+        :param source: The path to the text file
+        :param line_prefix: A regex to denote a directive (must be first thing in a line), which will
+            be immediately followed by the directive itself and optional arguments
+        """
+        TestConfig.__init__( self )
+
+        self.derive_config_from_text( source, line_prefix )
+        self.derive_tags_from_path( source )
+
+    def derive_config_from_text( self, source, line_prefix ):
+        regex = r'^' + line_prefix + r'(\S+)((?:\s+\S+)*?)\s*(?:#\s*(.*))?$'
+        for context in file.grep( regex, source ):
+            match = context['match']
+            directive = match.group( 1 )
+            text_params = match.group( 2 ).strip()
+            params = [s for s in text_params.split()]
+            comment = match.group( 3 )
+            if directive == 'device':
+                # log.d( '    configuration:', params )
+                if not params:
+                    log.e( source + '+' + str( context['index'] ) + ': device directive with no devices listed' )
+                elif 'each' in text_params.lower() and len( params ) > 1:
+                    log.e( source + '+' + str(
+                            context['index'] ) + ': each() cannot be used in combination with other specs', params )
+                elif 'each' in text_params.lower() and not re.fullmatch( r'each\(.+\)', text_params, re.IGNORECASE ):
+                    log.e( source + '+' + str( context['index'] ) + ': invalid \'each\' syntax:', params )
+                else:
+                    self._configurations.append( params )
+            elif directive == 'priority':
+                if len( params ) == 1 and params[0].isdigit():
+                    self._priority = int( params[0] )
+                else:
+                    log.e( source + '+' + str( context['index'] ) + ': priority directive with invalid parameters:',
+                           params )
+            elif directive == 'timeout':
+                if len( params ) == 1 and params[0].isdigit():
+                    self._timeout = int( params[0] )
+                else:
+                    log.e( source + '+' + str( context['index'] ) + ': timeout directive with invalid parameters:',
+                           params )
+            elif directive == 'tag':
+                self._tags.update( params )
+            elif directive == 'flag':
+                self._flags.update( params )
+            else:
+                log.e( source + '+' + str( context['index'] ) + ': invalid directive "' + directive + '"; ignoring' )
+
+    def derive_tags_from_path( self, source ):
+        # we need the relative ath starting at the unit-tests directory
+        relative_path = source.split( "unit-tests" + os.sep )[-1]
```

Comment:
What's this `"unit-tests" + `? It wasn't there in the original code, was it?

Reply:
Oh I see... you're taking everything after `unit-tests/`... hmm... what about a `/unit-tests/`, then?

Reply:
I need to use the relative path so I don't add tags from directories outside unit-tests. This line cuts the path so I only look at what's after `unit-tests/`

Reply:
Ok, I'll add an `os.sep` at the beginning.